src/main/java/com/adyen/service/TerminalCloudAPI.java

```diff
@@ -83,4 +83,24 @@ public TerminalAPIResponse sync(TerminalAPIRequest terminalAPIRequest) throws IO
         }.getType());
         return terminalAPIResponse;
     }
+
+    /**
+     * Checks if SaleToAcquirerData is set. If it is not creates an encoded base 64 string with application info.
+     *
+     * @param terminalAPIRequest TerminalAPIRequest
+     * @return TerminalAPIRequest
+     */
+    private static TerminalAPIRequest checkSaleToAcquirerDataApplicationInfo(TerminalAPIRequest terminalAPIRequest) {
+        if (terminalAPIRequest != null
+                && terminalAPIRequest.getSaleToPOIRequest() != null
+                && terminalAPIRequest.getSaleToPOIRequest().getPaymentRequest() != null
+                && terminalAPIRequest.getSaleToPOIRequest().getPaymentRequest().getSaleData() != null
+                && terminalAPIRequest.getSaleToPOIRequest().getPaymentRequest().getSaleData().getSaleToAcquirerData() == null) {
+            SaleToAcquirerDataModel saleToAcquirerDataModel = new SaleToAcquirerDataModel();
+            String saleToAcquirerDataJson = saleToAcquirerDataModel.toJson();
+            String saleToAcqurerDataJsonBase64 = Util.toBase64Encode(saleToAcquirerDataJson);
```

Comment:
would be nice to have that toBase64 method iplemented then you can use terminalAPIRequest.getSaleToPOIRequest().getPaymentRequest().getSaleData().setSaleToAcquirerData(saleToAcquirerDataModel.toBase64());